ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/turnstile/v1/turnstile/clientlibs/site/js/turnstilewidget.js

```diff
@@ -0,0 +1,93 @@
+/*******************************************************************************
+ * Copyright 2024 Adobe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+/**
+ * This class is responsible for interacting with the hCaptcha widget. It displays hCaptcha challenge.
+ */
+
+if (typeof window.TurnstileWidget === 'undefined') {
+    window.TurnstileWidget = class {
+        #widget = null
+        #model = null // passed by reference
+        #options = null
+        #lang = 'en'
+        static FD_CAPTCHA = "fd:captcha";
+
+        constructor(view, model, widget) {
+            // initialize the widget and model
+            this.#widget = widget;
+            this.#model = model;
+            this.#widget = document.createElement("div");
+            this.#widget.classList.add("cmp-adaptiveform-turnstile__widget");
+            this.#lang = view.formContainer.getModel().lang;
+
+            //Always inserting it in body
+            document.body.appendChild(this.#widget);
+            this.#options = Object.assign({}, this.#model._jsonModel);
+
+            this.#renderTurnstile(widget);
+        }
+
+        #renderTurnstile(element) {
+
+            var self = this;
+            var turnstileConfigData = this.#options;
+            element.innerHTML = '<div class="cf-turnstile"></div>';
+            var turnstileContainer = document.getElementsByClassName("cf-turnstile")[0];
+            var widgetId;
+            var url = turnstileConfigData.properties[TurnstileWidget.FD_CAPTCHA].config.uri;
+
+            var successCallback = function(response) {
```

Comment:
Please avoid using var and move to latest ES6 syntax, fix this at all places

Reply:
Done